pytact/data_reader.py

```diff
@@ -568,53 +568,43 @@ def get_index(index):
             return cast(Definition, Node(depgraph, lreader[depgraph].representative, lreader).definition)
         return TupleLike(count, get_index)
 
-    def _global_context(self, seen: set[Node]) -> Iterable[Definition]:
-        if prev := self.previous:
-            yield prev
-            yield from prev._global_context(seen)
-        for eprev in self.external_previous:
-            if eprev.node not in seen:
-                yield eprev
-                yield from eprev._global_context(seen)
-                seen.add(eprev.node)
-
-    def _clustered_global_context(self, seen: set[Node]) -> Iterable[list[Definition]]:
-        if prev := self.previous:
-            cluster = list(prev.cluster)
-            yield cluster
-            yield from cluster[-1]._clustered_global_context(seen)
-        for eprev in self.external_previous:
-            if eprev.node not in seen:
-                cluster = list(eprev.cluster)
-                yield cluster
-                yield from cluster[-1]._clustered_global_context(seen)
-                seen.add(eprev.node)
-
-    @property
-    def global_context(self) -> Iterable[Definition]:
+    def global_context(self, across_files : bool = True, inclusive = False, seen : set[Node] = None) -> Iterable[Definition]:
```

Comment:
Do you have a particular usecase in mind to use `seen` from the public API? If not, maybe this should be an auxiliary function and the real function doesn't have this argument?

Reply:
You are right, I just don't like auxiliary functions much :-), but you are probably right, let's make an auxiliary function with `seen`